examples/azure/chat_with_your_own_data.ipynb

```diff
@@ -0,0 +1,432 @@
+    "\n",
+    "Azure OpenAI on your data works with GPT-35-Turbo and GPT-4 language models, enabling them to provide responses based on your data. Grounding the model with data helps avoid outdated or incorrect responses."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## How it works\n",
+    "\n",
+    "[Azure OpenAI on your own data](https://learn.microsoft.com/azure/ai-services/openai/concepts/use-your-data) connects the model with your data, giving it the ability to retrieve and utilize data in a way that enhances the model's output. Together with Azure Cognitive Search, data is retrieved from designated data sources based on the user input and provided conversation history. The data is then augmented and resubmitted as a prompt to the model, giving the model contextual information it can use to generate a response.\n",
+    "\n",
+    "See the [Data, privacy, and security for Azure OpenAI Service](https://learn.microsoft.com/legal/cognitive-services/openai/data-privacy?context=%2Fazure%2Fai-services%2Fopenai%2Fcontext%2Fcontext) for more information."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Prerequisites\n",
+    "To get started, we'll cover a few prequisites. To use your own data with Azure OpenAI models, you will need:\n",
+    "\n",
+    "1. Azure OpenAI access and a resource with a chat model deployed (for example, GPT-3 or GPT-4)\n",
+    "2. Azure Cognitive Search resource\n",
+    "3. Azure Blob Storage resource\n",
+    "4. Your documents to be used as data (See [data source options](https://learn.microsoft.com/azure/ai-services/openai/concepts/use-your-data#data-source-options))\n",
+    "\n",
+    "\n",
+    "For a full walk-through on how to upload your documents to blob storage and create an index using the Azure AI Studio, see this [Quickstart](https://learn.microsoft.com/azure/ai-services/openai/use-your-data-quickstart?pivots=programming-language-studio&tabs=command-line).\n",
+    "\n",
+    "To connect with the search index, the following information is needed and represented as environment variables in this example:\n",
+    "\n",
+    "```\n",
+    "SEARCH_ENDPOINT\n",
+    "SEARCH_KEY\n",
+    "SEARCH_INDEX_NAME\n",
+    "```\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Setup\n",
+    "\n",
+    "First, we install the necessary dependencies."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install \"openai>=0.27.6\"\n",
+    "! pip install python-dotenv"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import openai\n",
+    "import dotenv\n",
+    "\n",
+    "dotenv.load_dotenv()"
```

Comment:
would be good to specify with a bit more detail what should go into the dotenv and maybe do it all in one step (as opposed to also talking about openai env variables below). 

an alternative approach could be to use something like `input` or `getpass` to prompt the user to input the variables

Reply:
Good point - I've included more detail on what needs to be added for dotenv